src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -124,21 +127,48 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
 
                 PrintProviders(providerCollection, enabledBy);
 
-                var process = Process.GetProcessById(processId);
+                DiagnosticsClient diagnosticsClient;
+                Process process;
+                if (ProcessLauncher.Launcher.HasChildProc)
+                {
+                    ReversedDiagnosticsClientBuilder builder = new ReversedDiagnosticsClientBuilder(ProcessLauncher.Launcher);
+                    try
+                    {
+                        diagnosticsClient = builder.Build(10);
+                    }
+                    catch (TimeoutException)
+                    {
+                        Console.Error.WriteLine("Unable to start tracing session - the target app failed to connect to the diagnostics transport. This may happen if the target application is running .NET Core 3.1 or older versions. Attaching at startup is only available from .NET 5.0 or later.");
```

Comment:
```suggestion
                        Console.Error.WriteLine("Unable to start tracing session - the target app failed to connect to the diagnostics port. This may happen if the target application is running .NET Core 3.1 or older versions. Attaching at startup is only available from .NET 5.0 or later.");
```